File: ts/2015/day1/day1.bench.ts
import { bench, describe } from "vitest";
import { getPositionToFirstEnterBasement, getPositionToFirstEnterBasementV2, getPositionToFirstEnterBasementV3, puzzleInput } from "./day1";


describe('', () => {
  bench('getPositionToFirstEnterBasement', () => {
    getPositionToFirstEnterBasement(puzzleInput)
  })

  bench('getPositionToFirstEnterBasementV2', () => {
    getPositionToFirstEnterBasementV2(puzzleInput)
  })

  bench('getPositionToFirstEnterBasementV3', () => {
    getPositionToFirstEnterBasementV3(puzzleInput)
  })
})

File: rb/2015/day1/day1_bench.rb
require_relative './day1'

require 'benchmark'

Benchmark.bm do |x|
  x.report("get_floor") { get_floor(puzzle_input) }
  x.report("get_floor_v2") { get_floor_v2(puzzle_input) }
end

Benchmark.bm do |x|
  x.report("get_position_to_first_enter_basement") { get_position_to_first_enter_basement(puzzle_input) }
  x.report("get_position_to_first_enter_basement_v2") { get_position_to_first_enter_basement_v2(puzzle_input) }
end